drivers/interrupt_controller/intc_intel_vtd.c

```diff
@@ -81,8 +82,14 @@ static int vtd_ictl_remap(const struct device *dev,
 
 	memset(&data->irte[irte_idx], 0, sizeof(struct vtd_irte));
 
+	/* TODO: SVT should be 01 for SDL and currect SID need to program*/
+
 	data->irte[irte_idx].l.vector = vector->arch.vector;
-	data->irte[irte_idx].l.dst_id = arch_curr_cpu()->id;
+	if (IS_ENABLED(CONFIG_X2APIC)) {
+		data->irte[irte_idx].l.dst_id = arch_curr_cpu()->id;
+	} else {
+		data->irte[irte_idx].l.dst_id = arch_curr_cpu()->id << 8;
+	}
```

Comment:
That's bogus (original code was well) arch_curr_cpu() does not return the CPU's logical APIC ID

Reply:
In xAPIC mode, APIC DestinationID is bit 47:40 which was not handled in existing code. We should ideally get the affinity info from device driver instead of taking from current CPU. I will create another PR for PCI host driver with API for interrupt affinity.

Reply:
That's not what I meant here. arch_curr_cpu()->id is not what we want. See https://github.com/zephyrproject-rtos/zephyr/pull/36731/commits/3f36187685d093b48989c0cbadd7e3db1373b500